packages/controllers/src/snaps/SnapController.ts

```diff
@@ -2052,14 +2056,48 @@ export class SnapController extends BaseController<
     snapId: SnapId,
     origin: string,
     request: Record<string, unknown>,
+  ): Promise<unknown> {
+    return this.handleRequest(snapId, origin, 'onRpcRequest', request);
+  }
+
+  /**
+   * Passes a JSON-RPC request object to the RPC handler function of a snap, triggering the onTxConfirmation handler.
+   *
+   * @param snapId - The ID of the recipient snap.
+   * @param origin - The origin of the RPC request.
+   * @param request - The JSON-RPC request object.
+   * @returns The result of the JSON-RPC request.
+   */
+  async handleTxInsightRequest(
+    snapId: SnapId,
+    origin: string,
+    request: Record<string, unknown>,
+  ): Promise<unknown> {
+    return this.handleRequest(snapId, origin, 'onTxConfirmation', request);
```

Comment:
I feel like there's a better name than `onTxConfirmation`. The confirmation is something of an implementation detail from the perspective of the snap. We should also specify that it's an Ethereum confirmation, and ideally refer to the purpose of the export, which is to suggest a tx insight.

`onEthereumTransaction` would probably be better, but it doesn't spark joy. Maybe `provideTransactionInsight`? `offerTx...`? `produce`? `create`? No obviously good answer.

Reply:
It's not necessarily an Ethereum confirmation, it could be any EVM-like chain. `provideTransactionInsight` might be the best idea so far, but it's not ideal. 😅